pkg/tests/apis/openapi_snapshots/API_CLIENT_GENERATION.md

```diff
@@ -0,0 +1,179 @@
+
+```jsx
+'../public/app/features/dashboard/api/endpoints.gen.ts': {
+    apiFile: '../public/app/features/dashboard/api/baseAPI.ts',
+    schemaFile: '../data/openapi/dashboard.grafana.app-v0alpha1.json',
+    apiImport: 'baseAPI',
+    filterEndpoints: ['createDashboard'],
+    argSuffix: 'Arg',
+    responseSuffix: 'Response',
+    tag: true,
+    hooks: true,
+},
+```
+
+### 5. Run the API Client script
+Then, we are ready to run the script to create the API client:
+
+```jsx
+yarn generate-apis
+```
+
+This will create an `endpoints.gen.ts` file in the path specified in the previous step.
+
+### 6. Create the index file for your hooks
+
+In the same `api` folder where the `endpoints.gen.ts` file has been saved, you have to create an index file from which you can import the types and hooks needed. By doing this, we selectively export hooks/types from `endpoints.gen.ts`.
+
+In our case, the dashboard index will be like:
+
+```jsx
+import { generatedAPI } from './endpoints.gen';
+
+export const dashboardAPI = generatedAPI;
+export const { useCreateDashboardMutation} = dashboardAPI;
+// eslint-disable-next-line no-barrel-files/no-barrel-files
+export { type Dashboard } from './endpoints.gen';
+
+```
+
+There are some use cases where the hook will not work, and that is a clue to see if it needs to be modified. The hooks can be tweaked by using `enhanceEndpoints`.
+
+TODO: Add examples of `enhanceEndpoints` usage
+
+### 7. Add information to the store
+
+The last, but not least, step to be done is adding the middleware and reducers to the store. You have available the oficial documentation in [RTK Query](https://redux-toolkit.js.org/tutorials/rtk-query#add-the-service-to-your-store)
+
+In Grafana, the reducers are added to `public/app/core/reducers/root.ts`:
+
+```jsx
+  const rootReducers = {
+    ...sharedReducers,
+    ...alertingReducers,
+    ...teamsReducers,
+    ...apiKeysReducers,
+    ...foldersReducers,
+    ...dashboardReducers,
+    ...exploreReducers,
+    ...dataSourcesReducers,
+    ...usersReducers,
+    ...serviceAccountsReducer,
+    ...userReducers,
+    ...invitesReducers,
+    ...organizationReducers,
+    ...browseDashboardsReducers,
+    ...ldapReducers,
+    ...importDashboardReducers,
+    ...panelEditorReducers,
+    ...panelsReducers,
+    ...templatingReducers,
+    ...supportBundlesReducer,
+    ...authConfigReducers,
+    plugins: pluginsReducer,
+    [alertingApi.reducerPath]: alertingApi.reducer,
+    [publicDashboardApi.reducerPath]: publicDashboardApi.reducer,
+    [browseDashboardsAPI.reducerPath]: browseDashboardsAPI.reducer,
+    [cloudMigrationAPI.reducerPath]: cloudMigrationAPI.reducer,
+    [iamApi.reducerPath]: iamApi.reducer,
+    [userPreferencesAPI.reducerPath]: userPreferencesAPI.reducer,
+    [provisioningAPI.reducerPath]: provisioningAPI.reducer,
+    [folderAPI.reducerPath]: folderAPI.reducer,
+    [dashboardAPI.reducerPath]: dashboardAPI.reducer,
+  };
+```
```

Comment:
i'd just really slim these examples down to only show the necessary info, e.g.

```suggestion
```jsx
  import { dashboardAPI } from '<pathToYourAPI>';
  const rootReducers = {
    ...,
    [dashboardAPI.reducerPath]: dashboardAPI.reducer,
  };
```
```